tests/pyccel/test_pyccel.py

```diff
@@ -82,26 +83,56 @@ def teardown(path_dir = None):
         elif not f.endswith(".py"):
             os.remove(file_name)
 
+def compare_pyth_fort_output_by_type( p_output, f_output, dtype=float ):
+
+    if dtype is str:
+        assert(p_output.strip()==f_output.strip())
+    elif dtype is complex:
+        p_output, f_output = compare_pyth_fort_output_by_type( p_output, f_output, float)
+        p_output, f_output = compare_pyth_fort_output_by_type( p_output, f_output, float)
+    elif dtype is bool:
+        rx = re.compile('TRUE|True|true|1|T|t|FALSE|False|false|F|f|0')
+        p_match = rx.search(p_output)
+        f_match = rx.search(f_output)
+        assert(p_match)
+        assert(f_match)
+        p = p_match.group().lower() in ['true', 't', '1']
+        f = f_match.group().lower() in ['true', 't', '1']
+        assert(p==f)
+        p_output=p_output[p_match.span()[1]:]
+        f_output=f_output[f_match.span()[1]:]
+
+    elif dtype is float or dtype is int:
+        if dtype is float:
+            rx = re.compile('[-0-9.eE]+')
+        elif dtype is int:
+            rx = re.compile('[-0-9eE]+')
```

Comment:
Why do you include the characters `eE` in the regular expression?

Reply:
I didn't at first but something was printing in scientific notation (ie 2.0E-02)

Reply:
Even the integers?

Reply:
I don't know. I thought it was better to put it in for things like 2e2 just in case rather than leave it out and have it fail later. But we can remove it if you don't think it's necessary

Reply:
You might be right, it depends on the string formatting that one uses after all. We will improve this in the future if need be.